collated/main/A0139498J.md

```diff
@@ -0,0 +1,1044 @@
+                lastShownList = model.getFilteredDoneTaskList();
+            } else {
+                lastShownList = model.getFilteredUndoneTaskList();
+            }
+    
+            if (lastShownList.size() < targetIndex - adjustmentForRemovedTask) {
+                continue;
+            }
+    
+            ReadOnlyTask taskToDelete = lastShownList.get(targetIndex - adjustmentForRemovedTask - 1);
+    
+            try {
+                model.deleteTask(taskToDelete);
+                deletedTasks.add((Task) taskToDelete);
+            } catch (TaskNotFoundException pnfe) {
+                assert false : "The target task cannot be missing";
+            }
+            displayDeletedTasks.add(taskToDelete.toString());
+            adjustmentForRemovedTask++;
+        }
+        
+       
+        updateHistory();
+        if (displayDeletedTasks.isEmpty()) {
+            indicateAttemptToExecuteIncorrectCommand();
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+        String toDisplay = displayDeletedTasks.toString().replace("[", "").replace("]", "");
+        return (displayDeletedTasks.size() == 1)? 
+                new CommandResult(String.format(MESSAGE_DELETE_ITEM_SUCCESS, toDisplay)):
+                new CommandResult(String.format(MESSAGE_DELETE_ITEMS_SUCCESS, toDisplay));
+    }
+
+```
+###### \java\seedu\address\logic\commands\DoneCommand.java
+``` java
+/**
+ * Archives a task identified using its last displayed index from the task manager.
+ */
+public class DoneCommand extends UndoableCommand {
+    
+    private static final Logger logger = LogsCenter.getLogger(DoneCommand.class);
+
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Archives the task identified by the index number used in the last task listing.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+    
+    public static final String TOOL_TIP = "done INDEX [ANOTHER_INDEX ...]";
+
+    public static final String MESSAGE_DONE_ITEM_SUCCESS = "Archived Task: %1$s";
+    public static final String MESSAGE_DONE_ITEMS_SUCCESS = "Archived Tasks: %1$s";
+    public static final String MESSAGE_DONE_UNDO_SUCCESS = "Undid archive tasks! Tasks restored to undone list!";
+    
+    
+    private List<Task> doneTasks;
+    private List<Task> readdedRecurringTasks;
+    private List<Task> doneTasksUndoFail;
+    private final List<Integer> targetIndexes;
+    private int adjustmentForRemovedTask = 0;
+    private boolean isViewingDoneList;
+
+    
+    public DoneCommand(List<Integer> targetIndexes) {
+        assert targetIndexes != null;
+        this.targetIndexes = targetIndexes;
+    }
+
+    @Override
+    public CommandResult execute() {
+        
+        assert model != null;
+        // check with Edmund, can rename to isRedoAction()
```

Comment:
remove such comments; you can use github comment feature for such communications between team members